### PR TITLE
feat: input to skip Trivy scan 

### PR DESCRIPTION
**Pull request by pankajmouriyakong**
Input: skip_trivy_scan
- This new input allows repo owner to choose whether to skip the scan or not.
- The default is set to 'false' to run the scan by default unless explicitly skipped by setting this to 'true'.
- Use `docker.io` instead of `ghcr.io` registry due to [intermittent issue](https://github.com/orgs/community/discussions/139074)

### CODE REVIEW

File: security-actions/scan-docker-image/action.yml [hunk 1]
```diff
@@ -296,7 +304,7 @@ runs:
 
     - name: Generate docker-cis JSON report
       uses: docker://ghcr.io/aquasecurity/trivy:0.37.2
-      if: ${{ steps.meta.outputs.scan_image != '' }}
+      if: ${{ inputs.skip_trivy_scan == 'false' && steps.meta.outputs.scan_image != '' }}
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:307]: Always use positive cases for better readability.

Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:307]:
```suggestion
      if: ${{ inputs.skip_trivy_scan != 'true' && steps.meta.outputs.scan_image != '' }}
```

File: security-actions/scan-docker-image/action.yml [hunk 2]
```diff
@@ -306,7 +314,7 @@ runs:
         input: ${{ steps.docker_tar.outputs.files_exists == 'true' && '--input' || '' }}
 
     - name: upload docker-cis JSON report
-      if: ${{ steps.meta.outputs.scan_image != '' }}
+      if: ${{ inputs.skip_trivy_scan == 'false' && steps.meta.outputs.scan_image != '' }}
```
Comment (saisatishkarra) [inline security-actions/scan-docker-image/action.yml:317]:
```suggestion
      if: ${{ inputs.skip_trivy_scan != 'true' && steps.meta.outputs.scan_image != '' }}
```

File: security-actions/scan-docker-image/action.yml (hunk 1 above)
Comment (pankajmouriyakong) [reply security-actions/scan-docker-image/action.yml]: got it

**Review by saisatishkarra — APPROVED** [review]
lgtm

**Pull request merged**